app/components/SideBar.js

```diff
@@ -1,51 +1,88 @@
-import React from 'react';
+import React, { Component } from 'react';
 import SchoolsNavTree from './NavTree';
 import _ from 'lodash';
 var classNames = require('classnames');
 import $ from 'jquery';
+import { connect } from 'react-redux';
+import { fetchEntitiesFromServer, toggleNode} from '../actions/';
+import PermissionsNavTree from './PermissionsNavTree';
+import * as Selectors from '../selectors/';
 
-
-var SideBar = React.createClass({
-  getInitialState() {
-    return {
+class SideBar extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
       isExpanded: false,
       results: []
     };
-  },
+  }
+
   componentDidMount() {
-  },
+  }
+
   /* Called when a component is reacting to a props change. Invoked before render is called. */
   componentWillReceiveProps(nextProps) {
-  },
+    console.log("In sidebar", nextProps.location);
+  }
 
-  toggleTree: function(e) {
+  onBoundaryClick(boundary) {
+    this.props.dispatch(toggleNode(boundary.id))
+    this.props.dispatch(fetchEntitiesFromServer(boundary.id));
+  }
+
+  toggleTree() {
     e.preventDefault();
     $("#wrapper").toggleClass("toggled");
-  },
+  }
 
-  render: function() {
-    let {onBoundaryClick, boundariesByParentId, boundaryDetails, primarySelected} = this.props;
+  render() {
+    let { boundariesByParentId, boundaryDetails, primarySelected, location } = this.props;
+    var DisplayElement;
     const schoolType = primarySelected ? 1 : 2
-    boundariesByParentId[1] = _.filter(boundariesByParentId[1], (key) =>  {
+    boundariesByParentId[1] = _.filter(boundariesByParentId[1], (key) => {
       const boundaryType = boundaryDetails[key].boundary_type
       return boundaryType ? boundaryType == schoolType : true
     })
     var sidebarClass = classNames({
       'toggled': this.state.isExpanded
     })
+    if(location.pathname.includes("permissions"))
```

Comment:
Ideally we'd want our UI to be a function of state in the store or localState. So one way to do that could be add a `navTreeType` to our redux store. Set NavTree type `onEnter` or `componentDidMount` on the `permissions` route. And again reset on the `boundary` routes.